### PR TITLE
control-service: protect against v1 cron batch not being supported

### PR DESCRIPTION
**Pull request by murphp15**
# Why
When we merged this PR https://github.com/vmware/versatile-data-kit/pull/1580/files we unintentionally broke compatibility with version 1.19 and 1.20.

We always check for v1 versions of batch jobs. checking for that version on clusters <1.21 will result in errors.

# What
Within this PR we add a guard around that statement and swallow the error if it happens. 

# How has this been tested
installed old version of k8s locally and ran integration tests against it. 
Within the class DataJobDeploymentCrudIT I also changed the property `atajobs.control.k8s.k8sSupportsV1CronJob` to `false`. As this is the config that should be set for 1.20.
All tests are green. 

I also ran the integration tests on main against the old cluster to make sure it catches the bug and it does result in many failed tests. 
 

# Not handled in this PR
How do we prevent issues like this in the future? 
Maybe integration tests run against many versions of k8s?

### CODE REVIEW

**Review by ivakoleva — APPROVED** [review]
What happens when another version pops up? May need to spend some time on designing a solution for addressing those compatibility issues better.

May you create and link a ticket for the mentioned in Testing done section?

**Comment by ivakoleva** [comment]
How did you test the change to verify the fix, for example locally?

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/KubernetesService.java [hunk 1]
```diff
@@ -786,15 +786,25 @@ public Set<String> listCronJobs() throws ApiException {
...
+              .listNamespacedCronJob(
+                  namespace, null, null, null, null, null, null, null, null, null, null);
+      v1CronJobNames =
+          v1CronJobs.getItems().stream()
+              .map(j -> j.getMetadata().getName())
+              .collect(Collectors.toSet());
+      log.debug("K8s V1 cron jobs: {}", v1CronJobNames);
+    } catch (ApiException e) {
+      if (e.getCode()
+          == 404) { // as soon as the minimum supported k8s version is >=1.21 then we should remove
```
[the remark this answers is not in the document]
Comment (murphp15) [reply projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/KubernetesService.java:801]: Its a good question and I don't know for 100% the answer. I think no.
But that is why I have left the comment to remove this clause as soon as we migrate to a newer version.

**Comment by murphp15** [comment]
@ivakoleva  
ticket describing testing problem https://github.com/vmware/versatile-data-kit/issues/1778.

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/KubernetesService.java [hunk 2]
```diff
@@ -786,15 +786,25 @@ public Set<String> listCronJobs() throws ApiException {
...
+                    // this.
+        log.debug("Unable to query for v1 batch jobs", e);
+      } else {
+        throw e;
+      }
+    }
 
     var v1BetaCronJobs =
```
[the remark this answers is not in the document]
Comment (murphp15) [reply projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/KubernetesService.java:809]: As mentioned in the comment here https://github.com/vmware/versatile-data-kit/pull/1777#discussion_r1146261091. 
There is some concern that we might swallow exceptions from other problems.
I want to keep this guard in place only on the section that will definitely throw it.

**Review by tozka — APPROVED** [review]

**Pull request merged**